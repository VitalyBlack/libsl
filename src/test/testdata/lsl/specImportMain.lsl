libsl "1.0.0";
library simple;

import "specImportSecondary";

automaton A : Int {
   var i: Int;

   fun f(param: Int) {
      action TEST_ACTION(1, "123", param);
      action TEST_ACTION_TWO();
   }
}