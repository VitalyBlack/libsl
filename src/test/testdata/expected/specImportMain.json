{
  "name": "simple",
  "lslVersion": "1.0.0",
  "automata": [
    {
      "name": "A",
      "type": "Int",
      "constructorVariables": [],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        }
      ],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "target": "A",
          "args": [
            {
              "name": "param",
              "type": "Int"
            }
          ],
          "contracts": [],
          "statements": [
            {
              "kind": "action",
              "actionName": "TEST_ACTION",
              "args": [
                {
                  "kind": "integer",
                  "value": 1
                },
                {
                  "kind": "string",
                  "value": "123"
                },
                {
                  "kind": "qualifiedAccess",
                  "access": {
                    "kind": "variableAccess",
                    "name": "param",
                    "variableInfo": {
                      "name": "param",
                      "fullName": "f.param",
                      "type": {
                        "name": "Int",
                        "isPointer": false,
                        "kind": "simple",
                        "realType": {
                          "name": "foo.bar.int",
                          "isPointer": false,
                          "kind": "real"
                        }
                      },
                      "kind": "functionArgument",
                      "function": "A.f",
                      "functionArgTypes": [
                        "Int"
                      ]
                    },
                    "type": "Int"
                  }
                }
              ]
            },
            {
              "kind": "action",
              "actionName": "TEST_ACTION_TWO",
              "args": []
            }
          ]
        }
      ]
    }
  ]
}